In [1]:
import numpy as np
import math
import matplotlib.pyplot as plt
from metpy.units import units

# A. Questions on Atmopsheric Scattering

# 1a. 
If the atmosphere did not scatter light, when you looked at the daytime sky, you would see black. Since particles in the atmopshere are responsible for scattering light upon contact with radiation, we see a blue sky during the day and red/orange/yellow sky in the evening. Without that, there would only be black, like in space.

# 1b.
$$
\lambda _{blue} = 450 nm
$$
$$
\lambda _{red} = 650 nm
$$
$$
I_{sca} \propto \lambda ^{-4}
$$
$$
I_{sca_{blue}} = 450 nm ^{-4}
$$
$$
I_{sca_{red}} = 650 nm ^{-4}
$$
$$
ratio blue to red = I_{sca_{blue}}/I_{sca_{red}}
$$

In [2]:
I_b = 450 ** -4
I_r = 650 ** -4

In [3]:
ratio_b_to_r = I_b/I_r

In [4]:
ratio_b_to_r

4.353147386069197

# ratio blue to red = 4.353147386069197

# range of UV light: https://www.ncbi.nlm.nih.gov/books/NBK304366/#:~:text=The%20wavelength%20of%20UV%20radiation,(100%E2%80%93280%20nm).

# selecting 200 nm as UV wavelength from above

$$
\lambda _{UV} = 200 nm
$$
$$
\lambda _{red} = 650 nm
$$
$$
I_{sca} \propto \lambda ^{-4}
$$
$$
I_{sca_{UV}} = 200 nm ^{-4}
$$
$$
I_{sca_{red}} = 650 nm ^{-4}
$$
$$
ratio UV to red = I_{sca_{UV}}/I_{sca_{red}}
$$

In [5]:
I_uv = 200 ** -4
I_r = 650 ** -4

In [6]:
ratio_uv_to_r = I_uv/I_r

In [7]:
ratio_uv_to_r

111.56640625000001

# ratio UV to red = 111.56640625000001

# 2.
At the top of the atmosphere there are very few particles/molecules/aerosols to cause scattering. Much like in the vacuum of space, which lacks particles/molecules/aerosols, the TOA appears black (no scattering). As you move down throuhg the atmosphere, Rayleigh scattering is responsible for the blue appearance. Rayleigh scattering is preferential when the responsible particles have a size parameter that is much smaller than incident wavelength. Blue light is prefferentially scattered, based off the scattering parameter (defined in the above question). Since blue light is preferentially scattered, there is more spectral radiance in the blue wavelength than violet, and since our eyes are more sensitive to blue light than violet, we observe blue light in the sky. Lower in the atmosphere, especially at sunrise/sunset, where light has to travel through a greater distance to reach our eyes at the surface, it has a red/orange appearance. This is due to the fact that blue light is scattered even more, and scattered away, giving red/orange light more chance to be scattered towards us, and leaving us with red/orange light. However, as you travel lower in the atmopshere, particle size gets bigger, as there are rain/cloud droplets and smoke/dust particles in the troposphere. At certain wavelengths, the scattering regime is Mie, rather than Rayleigh. This favors more colorful light reaching our eyes. At the surface, with larger objects like buildings and pavement, scattering is in the geometric regime. 

# 3.
The dashed line is the solar irradiance curve with respect to wavelength. More specifically, it is the blackbody solar irradiance curve for a temperature of 6000 K. The next solid line down is the solar irradiace curve outside the atmosphere, and the line below that is the solar irradiance curve at sea level for a zenith angle of 0 degrees. The shaded region between the bottom two curves represents absorption and the unshaded region between the two curves represents scattering. "Atmospheric windows" are wavelengths of light that are not absorbed by molecules in the atmosphere. In Earth's lower atmosphere, light is scattered by some part in the Rayleigh regime (most often blue light) and some in the Mie regime (larger particles such as rain drops in the lower atmosphere). Generally, we only consider scattering for solar radiation wavelengths (shortwave, .3-4.0 micrometers) and absorption for terrestrial radiation wavelengths (longwave, 3.5-50 micrometers.) This is because scattering is negligible for longer wavelengths and smaller particles. Additionally, for longer wavelengths, there is more absorption (less atmospheric windows) than scattering. At shorter wavelengths, there is much less absorption (more atmospheric windows) than scattering. Thus, as a general rule, we can consider scattering for solar radiation and absoprtion for terrestrial radiation.  